insights/core/spec_factory.py

```diff
@@ -712,11 +712,39 @@ def __call__(self, broker):
         ctx = _get_context(self.context, broker)
         p = os.path.join(ctx.root, self.path.lstrip('/'))
         p = ctx.locate_path(p)
-        result = sorted(os.listdir(p)) if os.path.isdir(p) else sorted(glob(p))
+        try:
+            result = os.listdir(p)
+        except OSError as e:
+            raise ContentException(str(e))
+        return sorted([r for r in result if not self.ignore_func(r)])
 
-        if result:
-            return [os.path.basename(r) for r in result if not self.ignore_func(r)]
-        raise ContentException("Can't list %s or nothing there." % p)
+    def _locate_path(self, broker):
```

Comment:
Do you want to remove this unused function?

Reply:
Indeed. Sorry for this omission and thanks for spotting it.